apps/api/src/routes/v1/graphql/+endpoint.ts

```diff
@@ -0,0 +1,80 @@
+import { join } from "node:path";
+import { parse } from "node:url";
+
+import { ApolloServer, HeaderMap, HTTPGraphQLRequest } from "@apollo/server";
+import {
+  ApolloServerPluginLandingPageLocalDefault,
+  ApolloServerPluginLandingPageProductionDefault,
+} from "@apollo/server/plugin/landingPage/default";
+import { loadFilesSync } from "@graphql-tools/load-files";
+import { mergeTypeDefs } from "@graphql-tools/merge";
+import { compress } from "@libs/lambda";
+import type { APIGatewayProxyHandler, APIGatewayProxyResult } from "aws-lambda";
+
+import { transformBody } from "./lib";
+import { resolvers } from "./resolvers";
+
+const graphqlServer = new ApolloServer({
+  introspection: true,
+  plugins: [
+    process.env.NODE_ENV === "development"
+      ? ApolloServerPluginLandingPageLocalDefault()
+      : ApolloServerPluginLandingPageProductionDefault({ footer: false }),
+  ],
+  resolvers,
+  typeDefs: mergeTypeDefs(loadFilesSync(join(__dirname, "schema/*.graphql"))),
+});
+
+export const ANY: APIGatewayProxyHandler = async (event) => {
+  const { body, headers: eventHeaders, httpMethod: method } = event;
+
+  try {
+    graphqlServer.assertStarted("");
+  } catch {
+    await graphqlServer.start();
+  }
+
+  const req: HTTPGraphQLRequest = {
+    body,
+    headers: Object.entries(eventHeaders).reduce(
+      (m, [k, v]) => m.set(k, Array.isArray(v) ? v.join(", ") : v ?? ""),
+      new HeaderMap(),
+    ),
+    method,
+    search: parse(event.path ?? "").search ?? "",
+  };
```

Comment:
Maybe this strategy could be worth checking out to make the object lighter? The idea is to lazily compute and cache the headers; i.e. only if they're actually accessed at least once during the request.

```suggestion
  let headers: Record<string, string>
  const req: HTTPGraphQLRequest = {
    body,
    get headers() {
      headers ??= Object.entries(eventHeaders).reduce(
        (m, [k, v]) => m.set(k, Array.isArray(v) ? v.join(", ") : v ?? ""),
        new HeaderMap(),
      )
      return headers
    },
    method,
    search: parse(event.path ?? "").search ?? "",
  };
```